src/features/vars-schema/env-vars.ts

```diff
@@ -56,16 +56,31 @@ export const INVALID_NEWLINE_REGEX = /\r|\n/;
 // Note that this list will be mutated by the cloud API to add
 // extra variables as for cloud-only features.
 export const SUPERVISOR_CONFIG_VAR_PROPERTIES: {
-	[k: string]: JSONSchema6Definition;
+	[k: string]:
+		| JSONSchema6Definition
+		| {
+				properties: {
+					required_reboot: boolean;
+					warning: string;
+				};
+		  };
 } = {
 	RESIN_SUPERVISOR_CONNECTIVITY_CHECK: {
 		enum: ['false', 'true'],
 		description: 'Enable / Disable VPN connectivity check',
+		properties: {
+			required_reboot: false,
+			warning: '',
+		},
 		default: 'true',
 	},
```

Comment:
I'm not sure how much discussion was hard regarding the schema for these object but I don't think having a properties object is that useful since the description and default keys are properties as well. Additionally, I think leaving the warning message to be handled by the UI makes more sense. What do you think about this:

```suggestion
	RESIN_SUPERVISOR_CONNECTIVITY_CHECK: {
		enum: ['false', 'true'],
		description: 'Enable / Disable VPN connectivity check',
		reboots: false
		default: 'true',
	},
```

Reply:
Agree with Miguel's suggestion, although maybe `requires_reboot` instead of `reboots`, but that's up to you @vip

Reply:
I didn't like `requires_reboot` because if I'm telling you something requires a reboot it's implying that you have to do the reboot after changing the value but the client nor the API do the rebooting so why say "requires".  You can just say the variable is going to reboot the device. `will_reboot` is another option.

Reply:
`will_reboot` sounds good too 👍🏼

Reply:
LOL, @thgreasi felt weird too about this. I'm glad more people are having the tingling sensation about this. I like `will_reboot` too. 

@cywang117 @20k-ultra Checkout the product call for decision on this, I have marked the timestamp from when you need to hear the call from. It's important in order for us to get on the same page: https://jel.ly.fish/brainstorm-topic-find-a-short-term-solution-for-notifying-users-that-a-config-variable-will-reboot-restart-the-device-ce3e30d0-879a-4bb7-a1e0-934d21dadda4

TLDR though: 
1. We went with having an additional properties property in order not to break JSONSchema and instead extend it. (check call for the discussion)
2. Both properties in the `properties` object are warnings. Check this out, 

IF you have `will_reboot: true` then UI handles the warning and shows a default message about the device/fleet going to reboot if you make this change. 

IF you have a `warning` message then UI shows the warning when this config variable is changed. These warnings are custom messages we want to give to the user about what's going to happen. Examples: You device might disconnect, you might blow up in 60 seconds, your device will now have a persistent journal that is going to take up space. 

IF both are there and present, then we show a composite warning. Merging both of them to be shown together.

Reply:
Thanks for the overview @vipulgupta2048.

The structure should then look like:

```suggestion
	RESIN_SUPERVISOR_CONNECTIVITY_CHECK: {
		enum: ['false', 'true'],
		description: 'Enable / Disable VPN connectivity check',
		default: 'true',
	},
```

If that config var triggered a reboot, then it should look like:

```suggestion
	RESIN_SUPERVISOR_CONNECTIVITY_CHECK: {
		enum: ['false', 'true'],
		description: 'Enable / Disable VPN connectivity check',
		properties: {
			requires_reboot: true,
		},
		default: 'true',
	},
```

If it should give out a warning when changed, then it should be:

```suggestion
	RESIN_SUPERVISOR_CONNECTIVITY_CHECK: {
		enum: ['false', 'true'],
		description: 'Enable / Disable VPN connectivity check',
		properties: {
			warning: 'Lorem ipsum ...',
		},
		default: 'true',
	},
```

The principle being that the absence of these properties cause no extra behaviour to be triggered.

> IF both are there and present, then we show a composite warning. Merging both of them to be shown together.

I don't understand what a composite warning is. Can you elaborate?

Reply:
Right on @dfunckt, Thodoris also gave me similar advice and I have gone for keeping the properties truthy and only present if things exist. Check out my latest commits for that.

The composite warning is shown when both are present and truthy. For example, consider these to be the values: 

```
will_reboot: true, 
warning: "Device is going to be hot."
```

Hence, we will show a "composite" of the 2 warnings in the UI. Something like. 

"The device will reboot when you make this change. 
Warning: The device is going to be hot."